### PR TITLE
Debug cleanup

### PR DESCRIPTION
**Pull request by CrashOveride95**
This PR is meant to clean up some of the debug support, and allow IS-Viewer to be easily started up without confusing patching to trick the internal IS-Viewer functions.
This additionally removes support for hardware that I guarantee nobody will be using with this library, including the Indy board and the Partner-N64.

`__osInitialize_autodetect` has been removed as a result, and `__osInitialize_common` has been renamed to `osInitialize`. A define has been added to `os_host.h` to keep compatibility with code that uses `__osInitialize_common` instead of the old `osInitialize` macro.

To use IS-Viewer now, simply add a call to `osInitialize_isv` in your boot function after you call `osInitialize`. After this, you can use `osSyncPrintf` calls to print messages to the debug console on emulators/flashcarts that emulate the IS-Viewer protocol.

This PR is being flagged as draft as I'd like to do more testing before I merge into `main` and the SDK repo

### CODE REVIEW

File: src/monegi/debug/assert.c [hunk 1]
```diff
@@ -1,5 +1,10 @@
 #include "PR/os_internal.h"
 
+extern void __assertBreak(void);
+#ifndef _FINALROM
+
 void __assert(const char* exp, const char* filename, int line) {
     osSyncPrintf("\nASSERTION FAULT: %s, %d: \"%s\"\n", filename, line, exp);
+	__assertBreak();
```
Comment (thecozies) [inline src/monegi/debug/assert.c]: mixing spaces and tabs 💀

File: src/monegi/debug/threadprofilereadtime.c [hunk 2]
```diff
@@ -49,3 +50,4 @@ OSTime osThreadProfileReadTimeTh(OSThread* thread) {
     }
     return thprof[id].time + adjust;
 }
+#endif
```
Comment (thecozies) [inline src/monegi/debug/threadprofilereadtime.c]: some of these files dont have the bonus line ending

**Review by thecozies — COMMENTED** [review]
last thing i'd recommend is providing header stubs for the stuff not compiled with `_FINALROM`

**Comment by ghost** [comment]
you should probably make an editorconfig for the spacing and newlines

**Comment by CrashOveride95** [comment]
I've been trying to figure out clang-format but it's so damn confusing

File: src/monegi/debug/assert.c (hunk 1 above)
Comment (CrashOveride95) [reply src/monegi/debug/assert.c]: Fixed in  https://github.com/ModernN64SDK/libultra_modern/pull/3/commits/88193bcfaa58475449a03a81c1ed0d015b70ffe8

File: src/monegi/debug/threadprofilereadtime.c (hunk 2 above)
Comment (CrashOveride95) [reply src/monegi/debug/threadprofilereadtime.c]: Fixed in  https://github.com/ModernN64SDK/libultra_modern/pull/3/commits/88193bcfaa58475449a03a81c1ed0d015b70ffe8

**Pull request merged**